code/modules/antagonists/arcfiend/arcfiend.dm

```diff
@@ -361,13 +361,26 @@ ABSTRACT_TYPE(/datum/targetable/arcfiend)
 
 	cast(atom/target)
 		. = ..()
-		if (target == holder.owner) return TRUE
 		if (!IN_RANGE(holder.owner, target, 1)) return TRUE
 		if (ishuman(target))
+			if (target == holder.owner)
+				self_cast(target)
+				return
 			actions.start(new/datum/action/bar/private/icon/jolt(holder.owner, target, holder, wattage), holder.owner)
 			logTheThing("combat", holder.owner, target, "[key_name(holder.owner)] used <b>[src.name]</b> on [key_name(target)] [log_loc(holder.owner)].")
 		else return TRUE
 
+	proc/self_cast(mob/living/carbon/human/self)
+		if (self.find_ailment_by_type(/datum/ailment/malady/flatline))
+			boutput(self, "<span class='alert'>You feel your heart jolt back into life!</span>")
+		else
+			boutput(self, "<span class='alert'>You feel a powerful jolt course through you!</span>")
+		playsound(self, "sound/effects/elec_bigzap.ogg", 30, 1)
```

Comment:
```suggestion
		playsound(self, 'sound/effects/elec_bigzap.ogg', 30, 1)
```
no need for sound cache, you arent string building

Reply:
oh I did not know about that caching behaviour, I just copied from a different arcfiend ability